runtime/src/common/quantity.rs

```diff
@@ -0,0 +1,75 @@
+//! An arbitrary precision unsigned integer.
+use std::fmt;
+
+use num_bigint::BigUint;
+use num_traits::{Num, Zero};
+use serde;
+
+/// An arbitrary precision unsigned integer.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
+pub struct Quantity(BigUint);
```

Comment:
Should probably add some test cases for quantity serialization/deserialization to ensure it matches go side?

Reply:
Good idea, added.